LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/Regions/Region.cs

```diff
@@ -217,7 +230,7 @@ public string GetDownstreamDR(Rxpk upstreamChannel, uint rx1DrOffset = 0)
                 if (rx1DrOffset <= this.RX1DROffsetTable.GetUpperBound(1))
                 {
                     // in case of EU, you respond on same frequency as you sent data.
-                    return this.DRtoConfiguration[(uint)this.RX1DROffsetTable[this.GetDRFromFreqAndChan(upstreamChannel.Datr), rx1DrOffset]].configuration;
+                    return this.DRtoConfiguration[(ushort)this.RX1DROffsetTable[this.GetDRFromFreqAndChan(upstreamChannel.Datr), rx1DrOffset]].configuration;

```

Comment:
Should we add log if for some reason the value we want is not found in the dictionary? This could make our life easier if it breaks here.